src/main/java/com/odysseusinc/arachne/portal/api/v1/controller/BaseStudyController.java

```diff
@@ -172,14 +181,36 @@ public BaseStudyController(BaseStudyService<T, DS, SS, SU> studyService,
                 T study = convert(studyDTO);
                 study = studyService.create(user, study);
                 result = new JsonResult<>(NO_ERROR);
-                result.setResult(convert(study));
+                result.setResult(convertStudyToStudyDTO(study));
             }
         } else {
             result = new JsonResult<>(PERMISSION_DENIED);
         }
         return result;
     }
 
+    @ApiOperation("Get existing or create new workspace.")
+    @RequestMapping(value = "/api/v1/workspace", method = GET)
+    public JsonResult<WD> getOrCreateWorkspace(Principal principal) throws PermissionDeniedException {
+
+        JsonResult<WD> result;
+        IUser user = getUser(principal);
+        T workspace = studyService.findOrCreateWorkspaceForUser(user.getId());
+        result = new JsonResult<>(NO_ERROR);
```

Comment:
this could be a simplier `return new JsonResult<>(NO_ERROR, convertStudyToWorkspaceDTO(workspace))`
and three lines of code shrims into one
just a comment, not requirement